src/main/java/hudson/remoting/Channel.java

```diff
@@ -1584,17 +1592,32 @@ public static Channel current() {
         return CURRENT.get();
     }
 
+    // TODO: Unrestrict after the merge into the master.
+    // By now one may use it via the reflection logic only
     /**
      * Calls {@link #dumpDiagnostics(PrintWriter)} across all the active channels in this system.
      * Used for diagnostics.
      *
-     * @since 2.26.3
+     * @param w Output destination
+     * 
+     * @since 2.62.3 - stable 2.x (restricted)
+     * @since TODO 3.x - public version 
      */
-    public static void dumpDiagnosticsForAll(PrintWriter w) throws IOException {
+    @Restricted(NoExternalUse.class)
+    public static void dumpDiagnosticsForAll(@Nonnull PrintWriter w) {
         for (Ref ref : ACTIVE_CHANNELS.values().toArray(new Ref[0])) {
             Channel ch = ref.channel();
-            if (ch!=null)
-                ch.dumpDiagnostics(w);
+            if (ch != null) {
+                try {
+                    ch.dumpDiagnostics(w);
```

Comment:
If we are performance concerned, I would recommend adding 

```
if (w.checkError()) {
  return; 
}
```

as no sense continuing if PrintWriter has already suppressed the IOE